docs/Compiled Braid Migration.md

```diff
@@ -0,0 +1,276 @@
+compared to the `vars` object imported from `braid-design-system/css`.
+If you depend on any of these extra properties and cannot find alternatives to replace them with,
+please reach out in the [#braid-support] Slack channel.
+
+### Component Prop Types
+
+Custom components that forward props to Braid components may need access to the types of some of the
+Braid component's props.
+These types should be derived from the component itself using the `ComponentProps` utility type from `react`.
+
+```diff
+// Error in Braid v32
+-import { TextProps } from 'braid-design-system/lib/components/Text/Text';
+// Derive props from `Text` instead
++import { Text } from 'braid-design-system';
++import { ComponentProps } from 'react';
+
++type TextProps = ComponentProps<typeof Text>;
+
+interface MyComponentProps {
+  tone: TextProps['tone'];
+}
+```
+
+#### Autosuggest props
+
+If your component wraps Braid's `Autosuggest` component, then you need to derive its props in a
+slightly different way.
+This is because `Autosuggest` is a generic component, so just using `ComponentProps<typeof Autosuggest>` will
+get you the prop type instantiated with the type `unknown`, effectively allowing any value to be
+passed into the `Autosuggest` :
+
+```tsx
+import { ComponentProps } from 'react';
+import { Autosuggest } from 'braid-design-system';
+
+type AutosuggestProps = ComponentProps<typeof Autosuggest>;
+
+// Equivalent to AutosuggestValue<unknown>
+type Value = AutosuggestProps['value'];
+
+const autosuggestValue: Value = {
+  text: 'myValue',
+  // value is of type unknown
+  value: ['this', 'can', 'be', 'literally', 'anything'],
+};
+
+// No errors!
+autosuggestValue.value = [1, 2, 3, 4, { whatever: 'you want' }];
+```
+
+The correct way to derive `Autosuggest`'s props is to use [typescript 4.7's instantiation expression feature][instantiation expression].
+This feature allows you to create a specialized version of a generic function that only accepts
+specific types. Since functional components are just functions, we can do this for components too:
+
+```tsx
+import { ComponentProps } from 'react';
+import { Autosuggest } from 'braid-design-system';
+
+// A specialized Autosuggest component that only accepts `string`s
+const StringAutosuggest = Autosuggest<string>;
+type AutosuggestProps = ComponentProps<typeof StringAutosuggest>;
+
+// Equivalent to AutosuggestValue<string>
+type Value = AutosuggestProps['value'];
+
+const autosuggestValue: Value = {
+  text: 'myValue',
+  // value is of type string
+  value: 'foo',
+};
+
+// Error: value must be of type string
+autosuggestValue.value = 123;
+```
+
+This has the obvious caveat that you must be on at least typescript 4.7 in order to use this
+feature.
+Your app will need to be on at least `sku@11.6.0` in order to use typescript 4.7.
+
+[instantiation expression]: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#instantiation-expressions
+
```

Comment:
```suggestion

#### Icon Slot Prop Type

If you wish to provide an icon slot on a component, you can derive the type of an icon component from an
existing Braid icon component:

```diff
// Error in Braid v32
-import { UseIconProps } from 'braid-design-system/lib/hooks/useIcon';
// Derive type from IconAdd instead
+import type { IconAdd } from 'braid-design-system';
+import { ComponentProps } from 'react';

+type IconProps = ComponentProps<typeof IconAdd>;

interface MyComponentProps {
-  icon?: FunctionComponent<UseIconProps>;
+  icon?: IconProps;
}

```
```